tests/integration_tests/synthetic_test_data.py

```diff
@@ -0,0 +1,71 @@
+"""Utilities for producing synthetic test data that is convergence-friendly."""
+
+from collections import namedtuple
+
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split
+
+RANDOM_SEED = 42
+NUMBER_OBSERVATIONS = 500
```

Comment:
Should we make this 100 to stay in line with what happens in most of the tests? They typically use 25 or 100 afaik

Reply:
Done.